text/0000-dockerfiles.md

```diff
@@ -0,0 +1,228 @@
+# Meta
+[meta]: #meta
+- Name: Support Dockerfiles
+- Start Date: 2021-06-30
+- Author(s): sclevine
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: [RFC0069](https://github.com/buildpacks/rfcs/blob/main/text/0069-stack-buildpacks.md), [RFC#167](https://github.com/buildpacks/rfcs/pull/167)
+- Depends on: [RFC#172](https://github.com/buildpacks/rfcs/pull/172)
+
+# Summary
+[summary]: #summary
+
+This RFC introduces functionality for customizing base images, as an alternative to stackpacks.
+
+# Motivation
+[motivation]: #motivation
+
+Relying on Dockerfiles for base image generation and manipulation allows us to apply buildpacks only to the problem that they solve best: managing application runtimes and dependencies.
+
+# What it is
+[what-it-is]: #what-it-is
+
+This RFC proposes that we replace stackpacks with build-time and runtime Dockerfiles that act as pre-build hooks.
+These hooks would be equivalent to base image / builder customizations, but would execute immediately before a build.
+
+Most importantly: the changes below do not impact buildpacks or the buildpack API.
+For a given application, a build that uses hooks could be optimized by creating a more narrowly-scoped builder with the hooks pre-applied, without modifying the buildpacks.
```

Comment:
This is correct. Nevertheless, this RFC should include within the section `What it is` a remark to mention that a dependency exists between what the Hook will do to extend the build image and/or run image and a buildpack.  
example: if the hook build Dockerfile will install a specific tool/lib (JDK, NodeJS, Ruby, Python,...), then the buildpack expects that the hook will extend the build layer of an existing image in order to add the tool/lib which is needed during the execution of the phase `bin/build`

Reply:
> if the hook build Dockerfile will install a specific tool/lib (JDK, NodeJS, Ruby, Python,...), then the buildpack expects that the hook will extend the build layer of an existing image in order to add the tool/lib which is needed during the execution of the phase bin/build

I don't think a buildpack expects that a *hook* will extend the builder -- just that the builder image has the correct tool/lib available before the build. If the hook is not present, but the builder image has the tool/lib installed, the buildpack would not know the difference.